src/plugins/index_pattern_field_editor/public/components/field_editor_flyout_content_container.tsx

```diff
@@ -0,0 +1,205 @@
+} from '../shared_imports';
+import { Field, PluginStart, InternalFieldType } from '../types';
+import { pluginName } from '../constants';
+import { deserializeField, getRuntimeFieldValidator } from '../lib';
+import { Props as FieldEditorProps } from './field_editor/field_editor';
+import { FieldEditorFlyoutContent } from './field_editor_flyout_content';
+
+export interface FieldEditorContext {
+  indexPattern: IndexPattern;
+  /**
+   * The Kibana field type of the field to create or edit
+   * Default: "runtime"
+   */
+  fieldTypeToProcess: InternalFieldType;
+  /** The search service from the data plugin */
+  search: DataPublicPluginStart['search'];
+}
+
+export interface Props {
+  /**
+   * Handler for the "save" footer button
+   */
+  onSave: (field: IndexPatternField) => void;
+  /**
+   * Handler for the "cancel" footer button
+   */
+  onCancel: () => void;
+  /**
+   * The docLinks start service from core
+   */
+  docLinks: DocLinksStart;
+  /**
+   * The context object specific to where the editor is currently being consumed
+   */
+  ctx: FieldEditorContext;
+  /**
+   * Optional field to edit
+   */
+  field?: IndexPatternField;
+  /**
+   * Services
+   */
+  indexPatternService: DataPublicPluginStart['indexPatterns'];
+  notifications: NotificationsStart;
+  fieldFormatEditors: PluginStart['fieldFormatEditors'];
+  fieldFormats: DataPublicPluginStart['fieldFormats'];
+  uiSettings: CoreStart['uiSettings'];
+  usageCollection: UsageCollectionStart;
+}
+
+/**
+ * The container component will be in charge of the communication with the index pattern service
+ * to retrieve/save the field in the saved object.
+ * The <FieldEditorFlyoutContent /> component is the presentational component that won't know
+ * anything about where a field comes from and where it should be persisted.
+ */
+
+export const FieldEditorFlyoutContentContainer = ({
+  field,
+  onSave,
+  onCancel,
+  docLinks,
+  indexPatternService,
+  ctx: { indexPattern, fieldTypeToProcess, search },
+  notifications,
+  fieldFormatEditors,
+  fieldFormats,
+  uiSettings,
+  usageCollection,
+}: Props) => {
+  const fieldToEdit = deserializeField(indexPattern, field);
+  const [Editor, setEditor] = useState<React.ComponentType<FieldEditorProps> | null>(null);
+  const [isSaving, setIsSaving] = useState(false);
+
+  const saveField = useCallback(
+    async (updatedField: Field) => {
+      setIsSaving(true);
+
+      const { script } = updatedField;
+
+      if (fieldTypeToProcess === 'runtime') {
+        try {
+          usageCollection.reportUiCounter(
+            pluginName,
+            usageCollection.METRIC_TYPE.COUNT,
+            'save_runtime'
+          );
+          // eslint-disable-next-line no-empty
+        } catch {}
+        // rename an existing runtime field
+        if (field?.name && field.name !== updatedField.name) {
+          indexPattern.removeRuntimeField(field.name);
+        }
+
+        indexPattern.addRuntimeField(updatedField.name, {
```

Comment:
Could it be a problem that, as I understand, we update-index pattern in memory but then if the update od saved object fails the we end up in an inconsistent intermediate state?

Reply:
I don't think the state is inconsistent. The user should be able to attempt saving again. This would be useful if there was an intermittent connection problem.